### PR TITLE
Nested Longhaul: Fix upstream protocol and test name

### PR DESCRIPTION
**Pull request by and-rewsmith**
Fixes:
- Upstream protocol is incorrectly configured. It assigns the correct upstream protocol, but always reports amqp to log analytics. I fixed this and standardized the naming for upstream protocol across our yamls as ```upstream.protocol```.
- The addition of test name I made has a problem, as I thought we had two nested longhaul pipelines. Turns out these are duplicates and we have one nested longhaul pipeline that runs two longhaul tests. One for amqp upstream and one for mqtt upstream with broker. Therefore we need to plumb this information down not through pipeline vars, but hardcoded values in the nested longhaul yaml similar to upstream protocol.

### CODE REVIEW

File: builds/e2e/templates/nested-longhaul-deploy.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ parameters:
   test.buildNumber: ''
   test.buildId: ''
   test.startDelay: ''
+  test.deploymentFileName: ''
```
Comment (and-rewsmith) [inline builds/e2e/templates/nested-longhaul-deploy.yaml:6]: moved for clarity

File: builds/e2e/nested-connectivity.yaml [hunk 2]
```diff
@@ -82,6 +82,7 @@ stages:
       identityServiceArtifactName: packages_ubuntu-18.04_amd64
       identityServicePackageFilter: aziot-identity-service_*_amd64.deb   
       artifactName: iotedged-ubuntu18.04-amd64
+      upstream.protocol: 'mqtt'
```
Comment (and-rewsmith) [inline builds/e2e/nested-connectivity.yaml:85]: This should be hardcoded, not fed through the pipeline. I tried to add a stage level environment variable for further consolidation (i.e. line 21), however devops wouldn't parse the yaml.

File: builds/e2e/templates/nested-parent-vm-setup.yaml [hunk 3]
```diff
@@ -29,20 +29,20 @@ jobs:
...
       artifactName: iotedged-ubuntu18.04-amd64
-      parentName: $[ dependencies.SetupVM_level5_${{ parameters.transportType }}.outputs['deployIoTEdge.deviceName'] ]
-      parentDeviceId: $[ dependencies.SetupVM_level5_${{ parameters.transportType }}.outputs['createIdentity.parentDeviceId'] ] 
-      deploymentFile: 'nestededge_middleLayerBaseDeployment_${{ parameters.transportType }}.json'
+      parentName: $[ dependencies.SetupVM_level5_${{ parameters['upstream.protocol'] }}.outputs['deployIoTEdge.deviceName'] ]
+      parentDeviceId: $[ dependencies.SetupVM_level5_${{ parameters['upstream.protocol'] }}.outputs['createIdentity.parentDeviceId'] ]
+      deploymentFile: "nestededge_middleLayerBaseDeployment_${{ parameters['upstream.protocol'] }}.json"
       level: 4
-      transportType: ${{ parameters.transportType }}
+      upstreamProtocol: ${{ parameters['upstream.protocol'] }}
```
Comment (and-rewsmith) [inline builds/e2e/templates/nested-parent-vm-setup.yaml:45]: This needs to be a bash var so cannot have dots.

**Comment by and-rewsmith** [comment]
Longhaul nested:
https://msazure.visualstudio.com/One/_build/results?buildId=40235061&view=results

Connectivity nested:
https://msazure.visualstudio.com/One/_build/results?buildId=40235088&view=results

E2E nested:
https://msazure.visualstudio.com/One/_build/results?buildId=40236107&view=results

**Review by dylanbronson — APPROVED** [review]
LGTM make sure all the pipelines are deploying correctly before pushing please

**Pull request merged**